identity/aziot-identityd/src/lib.rs

```diff
@@ -383,6 +383,18 @@ impl Api {
                     .await?
             }
             ReprovisionTrigger::Api => {
+                // Clear the backed up device state before reprovisioning.
+                // If this fails, log a warning but continue with reprovisioning.
+                let mut backup_file = self.settings.homedir.clone();
+                backup_file.push("device_info");
+
+                if let Err(err) = std::fs::remove_file(backup_file) {
+                    log::warn!(
```

Comment:
Need to treat it as successful if `err.kind() == std::io::ErrorKind::NotFound`

Reply:
Fixed